File: Backup/app/schemas/response/regex.py
from pydantic import Field
from typing import List, Any

from ..base import BaseResponse


class RegexTestResponse(BaseResponse):
    matches: list = Field(..., description="匹配结果")
    match_count: int = Field(..., description="匹配数量")
    match_details: List[dict] = Field(default=[], description="匹配详情")

File: Backup/tools_project_rust/src/routes/tools/regex.rs
use axum::{routing::post, Extension, Json, Router};

use crate::core::error::AppResult;
use crate::models::{BaseInfo, BaseResponse, RegexTestRequest, RegexTestResponse};
use crate::services::regex;

pub fn router() -> Router {
    Router::new().route("/test", post(test_regex))
}

async fn test_regex(
    Extension(base): Extension<Option<BaseInfo>>,
    Json(req): Json<RegexTestRequest>,
) -> AppResult<Json<BaseResponse<RegexTestResponse>>> {
    let (matches, match_details) = regex::test_regex(&req.pattern, &req.text, req.flags.as_deref())?;

    Ok(Json(BaseResponse::ok(
        RegexTestResponse {
            matches,
            match_count: match_details.len(),
            match_details,
        },
        base,
    )))
}
